src/checkers/inference/model/ConstraintManager.java

```diff
@@ -37,171 +30,186 @@
 
     private VisitorState visitorState;
 
-    private ConstraintVerifier constraintVerifier;
-
     public void init(InferenceAnnotatedTypeFactory inferenceTypeFactory) {
         this.inferenceTypeFactory = inferenceTypeFactory;
         this.realQualHierarchy = inferenceTypeFactory.getRealQualifierHierarchy();
         this.visitorState = inferenceTypeFactory.getVisitorState();
         this.checker = inferenceTypeFactory.getContext().getChecker();
-        this.constraintVerifier = new ConstraintVerifier(realQualHierarchy);
     }
 
     public Set<Constraint> getConstraints() {
         return constraints;
     }
 
-    public ConstraintVerifier getConstraintVerifier() {
-        return constraintVerifier;
-    }
-
-    private void add(Constraint constraint) {
-        if (!ignoreConstraints) {
+    /**
+     * Checks to see if the given {@link Constraint} is an instance of
+     * {@link AlwaysFalseConstraint}. If so, a warning is issued. If not, it adds the
+     * {@link Constraint} to the constraint set only if the {@code ignoreConstraints} flag is set to
+     * false and the constraint is not an {@link AlwaysTrueConstraint}.
+     *
+     * <p>
+     * The source constraint class is used to construct an error key for issue the warning.
+     *
+     * @param constraint a (possibly normalized) constraint
+     * @param nonNormalizedConstraintClass the class literal of the original non-normalized
+     *        constraint
+     */
+    private void add(Constraint constraint,
+            Class<? extends Constraint> nonNormalizedConstraintClass) {
+        if (constraint instanceof AlwaysFalseConstraint) {
+            // converts "XxxConstraint" into "xxx.constraint.unsatisfiable" for use as an error key
+            String errorKey = nonNormalizedConstraintClass.getSimpleName().toLowerCase()
+                    .replace("constraint", ".constraint.unsatisfiable");
+
+            // issue a non-halting error message indicating which non-normalized constraint cannot
+            // be satisfied
+            checker.report(Result.failure(errorKey), visitorState.getPath().getLeaf());
```

Comment:
Can we output more concrete message of unsatisfiablility, for example, `EqualityConstraint` between `@Top` and `@Bottom`. This is an issue before this PR, but it is definitely more helpful to output this message, too.

Reply:
And `checker.report(Result.failure()...` does cause inference to exit, if you call it `non-halting`.

Reply:
> Can we output more concrete message of unsatisfiablility, for example, EqualityConstraint between @Top and @Bottom. This is an issue before this PR, but it is definitely more helpful to output this message, too.

This method is shared across all constraints, so we'll need an output format for combine, existential, and preference constraint as well. They don't normalize for now, but that doesn't prevent a future refactoring from making them normalizable or for other >2 slot constraints normalizable. Suggestions?